### PR TITLE
Deploy

### PR DESCRIPTION
**Pull request by Luchy-B**
Requirements Met

- Splash screen
  -  A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  -  The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (categories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.
 
- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
      - [ ]  name (mandatory)
      - [ ]   amount (mandatory)
      - [ ]   categories (mandatory at least one)

  - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.

### CODE REVIEW

File: spec/requests/operations_spec.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe '/', type: :request do
+  context 'GET /' do
+    it 'renders a successful response' do
+      get root_path
+      expect(response).to have_http_status(200)
+    end
+  end
+end
```
Comment (mukhammadrizooff) [inline spec/requests/operations_spec.rb]: - [ ] [REQUIRED] Good job so far! Kindly, this part `failed` with  `ActionView::Template::Error:`, please fix this `spec` 🙏🏼 

image

**Review by mukhammadrizooff — CHANGES_REQUESTED** [review]
Hi @Luchy-B, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Required Changes ♻️
- [ ] [REQUIRED] Good job so far! I appraciate what you have done but your `button`("New Category", ...) and `navigation`("Most recent", ...), colors I mentioned below are not matching with [Behance Design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), please implement them 🙏🏼 

image


image



_Check the comments under the review._

## Optional suggestions
- [ ] [OPTIONAL] It would be more professional if you implemen your `GitFlow` like that(`main to dev`) 👍🏼

image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/operations_spec.rb (hunk 1 above)
Comment (Luchy-B) [reply spec/requests/operations_spec.rb]: Thank you for the review. Test fixed

Comment (Whoistolu) [reply spec/requests/operations_spec.rb]: - You have a done a great job resolving the test.

File: app/views/categories/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  <% end %>
+
+  <div class = "footer_container">
+ <div class ="footer">
+ <p class="forgot-pwd"><%= link_to "ADD NEW CATEGORY", new_category_path(@user) %></p>
+ </div>
+ </div>
+
+  
+</div>
```
Comment (Whoistolu) [inline app/views/categories/index.html.erb]: - [ ] [REQUIRED] Kindly check the recommendation the previous reviewer dropped on the review summary where he stated that you need to stick to the Behance design color, typography: font face, size, and weight, Layout: composition, and space between elements. Kindly ensure that the nav bar, the Add new category button, and the application background has the same color as recommended in the Behance design. Kindly implement this for the rest of the pages as well to enhance uniformity.
Take a look at this:

Your Implementation   |  Figma Design
:-------------------------:|:-------------------------:
Screenshot 2023-07-19 at 22 52 49 | Screenshot 2023-07-19 at 23 02 49

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @Luchy-B 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @Luchy-B,

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Luchy-B** [comment]
Thank you for the review

**Pull request merged**